posthog/tasks/calculate_event_property_usage.py

```diff
@@ -220,3 +230,33 @@ def _get_event_properties(team_id: int, since: timezone.datetime) -> List[Tuple[
     from posthog.models.event.sql import GET_EVENT_PROPERTIES
 
     return sync_execute(GET_EVENT_PROPERTIES, {"team_id": team_id, "timestamp": since})
+
+
+def _get_insight_query_usage(team_id: int, since: datetime) -> Tuple[List[str], List[str]]:
+    event_usage: List[str] = []
+    property_usage: List[str] = []
+
+    insight_filters = [
+        (id, Filter(data=filters) if filters else None)
+        for (id, filters) in Insight.objects.filter(team__id=team_id, created_at__gt=since)
+        .values_list("id", "filters")
+        .all()
+    ]
+    for id, item_filters in insight_filters:
+        if item_filters is None:
+            logger.info(
+                "calculate_event_property_usage_for_team.insight_has_no_filters",
+                team=team_id,
+                insight_id=id,
+            )
+            continue
+
+        for item_filter_event in item_filters.events:
+            event_usage.append(str(item_filter_event.id))
+            property_usage.extend([p.key for p in item_filter_event.property_groups.flat])
+
+        property_usage.extend([p.key for p in item_filters.property_groups.flat])
+
+        # todo are there more usages of properties? e.g. math aggregations?
```

Comment:
Yep, there's quite a few, here's a full list: It might be easier to just use the `ColumnOptimizer` here? Ensures that all future new properties will be counted as well :) 
 https://github.com/PostHog/posthog/blob/master/posthog/queries/column_optimizer/foss_column_optimizer.py#L102

Reply:
\o/ awesome

Reply:
@neilkakkar that was a great shout. Other tests started reporting counted properties where previously they were reporting None/0

I think this is in a place to merge and see what happens in prod. The benefit of working on something already broken... 🤣

Reply:
Noice!

One thing I'm unfamiliar with: Is this supposed to count event & person properties both? `properties_used_with_filter` returns a tuple of property, table, count. It's possible that the property you want is the event one, but the one being used is the person one.

If you only want event properties, use `_used_properties_with_type`, like so: https://github.com/PostHog/posthog/blob/master/posthog/queries/column_optimizer/foss_column_optimizer.py#L34

Reply:
Ah, yes, great catch!